Update 1.7.2

blmaier opened the pull request:
<!--
Thank you for your pull request.
Below are a few things we ask you kindly to self-check before getting a review. Remove checks that are not relevant.
-->
Checklist
* [x] Used a [personal fork of the feedstock to propose changes](https://conda-forge.org/docs/maintainer/updating_pkgs.html#forking-and-pull-requests)
* [x] Bumped the build number (if the version is unchanged)
* [x] Reset the build number to `0` (if the version changed)
* [x] [Re-rendered]( https://conda-forge.org/docs/maintainer/updating_pkgs.html#rerendering-feedstocks ) with the latest `conda-smithy` (Use the phrase <code>@<space/>conda-forge-admin, please rerender</code> in a comment in this PR for automated rerendering)
* [x] Ensured the license file is being packaged.

<!--
Please note any issues this fixes using [closing keywords]( https://help.github.com/articles/closing-issues-using-keywords/ ):
-->

<!--
Please add any other relevant info below:
-->

blmaier committed:
update to 1.7.2

blmaier committed:
MNT: Re-rendered with conda-build 24.5.1, conda-smithy 3.44.5, and conda-forge-pinning 2024.11.16.06.56.49

conda-forge-admin commented:
Hi! This is the friendly automated conda-forge-linting service.

I just wanted to let you know that I linted all conda-recipes in your PR (```recipe/meta.yaml```) and found it was in an excellent condition.

blmaier committed:
add setuptools to package

blmaier committed:
move setuptools to host

blmaier committed:
fix bug in python tests

The latest version of swig 4.3 introduced a backwards compatibility
issue that is causing the python tests to fail

conda-forge-admin commented:
Hi! This is the friendly automated conda-forge-linting service.

I wanted to let you know that I linted all conda-recipes in your PR (```recipe/meta.yaml```) and found some lint.

Here's what I've got...


For **recipe/meta.yaml**:

 * ❌ ``requirements: build: swig<4.2`` must contain a space between the name and the pin, i.e. ``swig <4.2``

conda-forge-admin commented:
Hi! This is the friendly automated conda-forge-linting service.

I just wanted to let you know that I linted all conda-recipes in your PR (```recipe/meta.yaml```) and found it was in an excellent condition.

blmaier committed:
fix test for shared library name

blmaier committed:
fix missing libfdt.so.1 in pylibfdt

blmaier committed:
MNT: Re-rendered with conda-build 24.5.1, conda-smithy 3.44.5, and conda-forge-pinning 2024.11.16.14.10.20

blmaier committed:
fix build warning of missing pkg-config during install

blmaier committed:
try to fix error from pylibfdt install

> Couldn't find index page for 'libfdt'

blmaier committed:
Revert "try to fix error from pylibfdt install"

This reverts commit d1fa089b8516cd3d902d4d345e2773308d8e5fca.

blmaier committed:
port to Meson

blmaier committed:
add MESON_ARGS to setup

blmaier committed:
fix tests building on linux

blmaier committed:
set DESTDIR for build

blmaier committed:
disable tests

blmaier committed:
remove subpackage unit tests

We run the unit-tests during the build. Do not run them on the
sub-packages as it overcomplicates packaging.

blmaier committed:
specify files without include:

blmaier committed:
fix DESTDIR installation issue

We can't install with DESTDIR set to PREFIX as installers expect to
install to $DESTDIR/$PREFIX.

blmaier committed:
remove unused build requirements

blmaier committed:
tighten scope of pylibfdt files

blmaier committed:
add missing libyaml during test

blmaier committed:
move meson to host directory

blmaier committed:
migrate python install to pip

blmaier committed:
migrate install to output/scripts

conda-forge-admin commented:
Hi! This is the friendly automated conda-forge-linting service.

I failed to even lint the recipe, probably because of a conda-smithy bug :cry:. This likely indicates a problem in your `meta.yaml`, though. To get a traceback to help figure out what's going on, install conda-smithy and run `conda smithy recipe-lint --conda-forge .` from the recipe directory. You can also examine the [workflow logs](https://github.com/conda-forge/conda-forge-webservices/actions/runs/11988767415) for more detail.


<sub>This message was generated by GitHub Actions workflow run [https://github.com/conda-forge/conda-forge-webservices/actions/runs/11988767415](https://github.com/conda-forge/conda-forge-webservices/actions/runs/11988767415). Examine the logs at this URL for more detail.</sub>

blmaier committed:
move python install to seperate script

blmaier committed:
fix meta typo

conda-forge-admin commented:
Hi! This is the friendly automated conda-forge-linting service.

I just wanted to let you know that I linted all conda-recipes in your PR (```recipe/meta.yaml```) and found it was in an excellent condition.

blmaier committed:
add all build tools to outputs

blmaier committed:
add missing pip package

blmaier committed:
fix osx library names

blmaier committed:
MNT: Re-rendered with conda-build 24.5.1, conda-smithy 3.44.7, and conda-forge-pinning 2024.11.23.13.45.45

blmaier committed:
fix file include list

blmaier committed:
remove unused patch

blmaier committed:
enable full tests on linux

blmaier committed:
Revert "enable full tests on linux"

This reverts commit a496233643dc80078ea55cf77adde363aa342526.

blmaier committed:
remove valgrind

blmaier committed:
Reapply "enable full tests on linux"

This reverts commit c39ac06720efb9b4fa55a6ab8547caea4a83106a.

blmaier committed:
fix error building tests

blmaier committed:
fix patch directory

blmaier committed:
remove old build comment

blmaier committed:
add upstream link to patch

blmaier committed:
remove unneeded build number

blmaier committed:
make c compiler comment consistent

blmaier commented:
Hi @conda-forge/core team, this package appears unmaintained. Could I get a review on this update?

h-vetinari commented on recipe/meta.yaml:
```diff
@@ -36,22 +35,26 @@ outputs:
+        - pkg-config
       host:
         - {{ pin_subpackage("libfdt", exact=True) }}
+        - python {{ python }}
```
Why is python needed here? The binaries are compiled from C, which is confirmed by the link check:
```
   INFO (dtc,bin/fdtoverlay): Needed DSO lib/libfdt.so.1 found in local/linux-64::libfdt==1.7.2=h9ecbd09_0
   INFO (dtc,bin/fdtoverlay): Needed DSO x86_64-conda-linux-gnu/sysroot/lib64/libc.so.6 found in CDT/compiler package conda-forge/noarch::sysroot_linux-64==2.17=h4a8ded7_18
   INFO (dtc,bin/fdtput): Needed DSO lib/libfdt.so.1 found in local/linux-64::libfdt==1.7.2=h9ecbd09_0
   INFO (dtc,bin/fdtput): Needed DSO x86_64-conda-linux-gnu/sysroot/lib64/libc.so.6 found in CDT/compiler package conda-forge/noarch::sysroot_linux-64==2.17=h4a8ded7_18
   INFO (dtc,bin/fdtdump): Needed DSO x86_64-conda-linux-gnu/sysroot/lib64/libc.so.6 found in CDT/compiler package conda-forge/noarch::sysroot_linux-64==2.17=h4a8ded7_18
   INFO (dtc,bin/convert-dtsv0): Needed DSO x86_64-conda-linux-gnu/sysroot/lib64/libc.so.6 found in CDT/compiler package conda-forge/noarch::sysroot_linux-64==2.17=h4a8ded7_18
   INFO (dtc,bin/fdtget): Needed DSO lib/libfdt.so.1 found in local/linux-64::libfdt==1.7.2=h9ecbd09_0
   INFO (dtc,bin/fdtget): Needed DSO x86_64-conda-linux-gnu/sysroot/lib64/libc.so.6 found in CDT/compiler package conda-forge/noarch::sysroot_linux-64==2.17=h4a8ded7_18
   INFO (dtc,bin/dtc): Needed DSO lib/libyaml-0.so.2 found in conda-forge/linux-64::yaml==0.2.5=h7f98852_2
   INFO (dtc,bin/dtc): Needed DSO x86_64-conda-linux-gnu/sysroot/lib64/libc.so.6 found in CDT/compiler package conda-forge/noarch::sysroot_linux-64==2.17=h4a8ded7_18
```
I think this should just be removed
```suggestion
```

blmaier replied:
I agree both the dtc and libfdt outputs shouldn't need python. I had tried removing this before, and when doing so the templating can't find PYTHON. Do you know why that would be?

https://dev.azure.com/conda-forge/feedstock-builds/_build/results?buildId=1089416&view=logs&j=d818e662-7181-5ce9-b265-50d03742b7af&t=367317f3-9b77-59ac-9e12-e56370c98acd&l=607

h-vetinari commented on recipe/meta.yaml:
```diff
@@ -64,62 +67,44 @@ outputs:
-        - make
-        - swig
+        - {{ stdlib('c') }}
+        - swig <4.2
```
Can you comment why the cap on swig is necessary?

blmaier replied:
I was hitting some bug with swig that I fixed by switching to 4.2, but it builds with swig 4.3 now. Removed the limit.

h-vetinari commented on recipe/meta.yaml:
```diff
@@ -131,26 +116,39 @@ outputs:
-        - test -f $PREFIX/lib/libfdt${SHLIB_EXT}.{{ major_ver }}
+        - test -f $PREFIX/lib/libfdt.{{ major_ver }}${SHLIB_EXT}  # [osx]
+        - test -f $PREFIX/lib/libfdt${SHLIB_EXT}.{{ major_ver }}  # [linux]
+        - test -f $PREFIX/lib/libfdt${SHLIB_EXT}.{{ version }}  # [linux]
```
I'm assuming the switch to the more standard SOVER-handling on osx is due to the switch to meson. Given that `libfdt` advertises a minor-level pin in the run-export, and we're going from 1.6 to 1.7, this should be fine. For completeness, I check who uses `libfdt`, and it's actually only used by this feedstock itself.
```
$mamba repoquery whoneeds -c conda-forge -p linux-64 libfdt

Executing the query libfdt

conda-forge/noarch                                  17.4MB @   1.5MB/s 11.4s
conda-forge/linux-64                                40.0MB @   2.6MB/s 15.7s


 Name Version Build        Depends     Channel  Subdir
-------------------------------------------------------
 dtc  1.6.1   h166bdaf_0   conda-forge linux-64
 dtc  1.6.1   h166bdaf_1   conda-forge linux-64
 dtc  1.6.1   h166bdaf_2   conda-forge linux-64
```

blmaier replied:
Thank you for verifying that. How did you check the run-export?

h-vetinari replied:
See here
https://github.com/conda-forge/dtc-package-feedstock/blob/dff90d43a55dea85754ecca9472c7aa3094c58bf/recipe/meta.yaml#L129-L130

h-vetinari reviewed (commented):
This looks very good, thanks a lot for stepping up @blmaier! Indeed, it looks like @timsnyder hasn't had any activity in conda-forge since ~March. The only thing I think we need to fix is that the binaries in `dtc` don't need (and shouldn't have) a python dependence.

blmaier committed:
remove host python from dtc

Co-authored-by: h-vetinari <h.vetinari@gmx.com>

h-vetinari committed:
use python directly in build invocation

h-vetinari committed:
remove python dependency also from libfdt

h-vetinari commented:
@conda-forge-admin, please rerender

conda-forge-admin commented:
Hi! This is the friendly automated conda-forge-webservice.

I tried to rerender for you, but it looks like there was nothing to do.


<sub>This message was generated by GitHub Actions workflow run [https://github.com/conda-forge/conda-forge-webservices/actions/runs/11991831312](https://github.com/conda-forge/conda-forge-webservices/actions/runs/11991831312). Examine the logs at this URL for more detail.</sub>

blmaier committed:
remove restriction on swig <4.2

h-vetinari commented:
One thing that's a bit silly is that each job builds the same `libfdt` and `dtc`
```
# job for 3.9
anaconda upload \
    /home/conda/feedstock_root/build_artifacts/linux-64/libfdt-1.7.2-hb9d3cd8_0.conda \
    /home/conda/feedstock_root/build_artifacts/linux-64/dtc-1.7.2-hb9d3cd8_0.conda \
    /home/conda/feedstock_root/build_artifacts/linux-64/pylibfdt-1.7.2-py39h8cd3c5a_0.conda
# 3.10
anaconda upload \
    /home/conda/feedstock_root/build_artifacts/linux-64/libfdt-1.7.2-hb9d3cd8_0.conda \
    /home/conda/feedstock_root/build_artifacts/linux-64/dtc-1.7.2-hb9d3cd8_0.conda \
    /home/conda/feedstock_root/build_artifacts/linux-64/pylibfdt-1.7.2-py310ha75aee5_0.conda
# 3.11
anaconda upload \
    /home/conda/feedstock_root/build_artifacts/linux-64/libfdt-1.7.2-hb9d3cd8_0.conda \
    /home/conda/feedstock_root/build_artifacts/linux-64/dtc-1.7.2-hb9d3cd8_0.conda \
    /home/conda/feedstock_root/build_artifacts/linux-64/pylibfdt-1.7.2-py311h9ecbd09_0.conda
# 3.12
anaconda upload \
    /home/conda/feedstock_root/build_artifacts/linux-64/libfdt-1.7.2-hb9d3cd8_0.conda \
    /home/conda/feedstock_root/build_artifacts/linux-64/dtc-1.7.2-hb9d3cd8_0.conda \
    /home/conda/feedstock_root/build_artifacts/linux-64/pylibfdt-1.7.2-py312h66e93f0_0.conda
```
This is not the end of the world, in the sense that all of them have exactly colliding build strings, meaning it'll only get uploaded once (by whichever job finishes first).

What would be better would be to collapse them all into a single job per platform that builds the lib and then 4 x `pylibfdt`. Given that he build scripts are already separated here, that should actually be pretty easy.

h-vetinari committed:
set feedstock-name

h-vetinari committed:
add blmaier to maintainers

h-vetinari committed:
remove python from global requirements (for `build.sh`)

h-vetinari commented:
@conda-forge-admin, please rerender

Commit:
MNT: Re-rendered with conda-build 24.9.0, conda-smithy 3.44.7, and conda-forge-pinning 2024.11.23.20.32.37

h-vetinari commented:
> One thing that's a bit silly is that each job builds the same `libfdt` and `dtc`

Contrast to all-in-one after updates 🙃 
```
anaconda upload \
    /home/conda/feedstock_root/build_artifacts/linux-64/libfdt-1.7.2-hb9d3cd8_0.conda \
    /home/conda/feedstock_root/build_artifacts/linux-64/dtc-1.7.2-hb9d3cd8_0.conda \
    /home/conda/feedstock_root/build_artifacts/linux-64/pylibfdt-1.7.2-py312h66e93f0_0.conda \
    /home/conda/feedstock_root/build_artifacts/linux-64/pylibfdt-1.7.2-py310ha75aee5_0.conda \
    /home/conda/feedstock_root/build_artifacts/linux-64/pylibfdt-1.7.2-py39h8cd3c5a_0.conda \
    /home/conda/feedstock_root/build_artifacts/linux-64/pylibfdt-1.7.2-py311h9ecbd09_0.conda
```

h-vetinari reviewed (approved):
Thank you for stepping up! 💪🙏

blmaier commented:
Thanks for consolidating the build outputs, that's a nice improvement 😄. Thank you for reviewing my PR
